star-rx72n-firmware/lib/rx_bq4050/src/rx_bq4050.c

```
@@ -262,13 +219,19 @@ rx_bq4050_read_relative_soc(rx_bus_manager_t* manager, const char* bus_name, uin
   RX_CHECK_NULL_PTR(soc_percent, s_tag, "soc_percent pointer is NULL");
 
-  uint16_t soc_word;
-  rx_err_t err =
+  uint16_t       soc_word;
+  const rx_err_t err =
     rx_bus_smbus_read_word_data(manager, bus_name, k_sbs_relative_state_of_charge, &soc_word);
 
   if (err == k_rx_ok) {
     /* Clamp to 0-100 range (SBS allows > 100% in some conditions) */
-    *soc_percent = (soc_word > k_soc_max_percent) ? k_soc_max_percent : (uint8_t)soc_word;
+    if (soc_word < k_soc_min_percent) {
```

Clang (14.0.6)

[warning] repeated branch in conditional chain

(bugprone-branch-clone)